### DISCUSSION TITLE
FileStream.FlushAsync(true), can it exist?

### DISCUSSION THREAD

**Discussion by jhudsoncedaron** [dotnet/runtime#62555] [Ideas] [answered]
While we can't completely become async this idea of push everything large to async is still kicking around in our codebase. On attempting one, discovered a case where FileSream.Flush(true) is called; would need an async version as that's where the actual jam is.

I don't know if this even can exist.

**Comment by gfoidl** [comment]
Do you mean [FileStream.FlushAsync](https://docs.microsoft.com/en-us/dotnet/api/system.io.filestream.flushasync?view=net-6.0)?
It's already around for some time.

**Comment by jhudsoncedaron** [comment]
@gfoidl: Definitely not. FileStream.Flush(true) is a special function not on stream that does not return until the block is written all the way through to the disk.

**Comment by svick** [comment, marked as answer]
This was previously discussed at https://github.com/dotnet/runtime/issues/28444, where @stephentoub said:

> And as to my knowledge there's no asynchronous equivalent at the OS level exposed for flushToDisk==true, I wouldn't want to expose a FlushAsync(true)… if a developer wants to Task.Run(() => Flush(true)), they can do so.
